gcc/rust/hir/rust-ast-lower-pattern.cc

```diff
@@ -333,9 +333,13 @@ ASTLoweringPattern::visit (AST::AltPattern &pattern)
     = new HIR::AltPattern (mapping, std::move (alts), pattern.get_locus ());
 
   if (is_let_top_level)
-    rust_error_at (pattern.get_locus (),
-		   "top level alternate patterns are not allowed for %<let%> "
-		   "bindings - use an outer grouped pattern");
+    {
+      rich_location richloc (line_table, pattern.get_locus ());
+      richloc.add_fixit_replace ("use an outer grouped pattern");
+      rust_error_at (
+	richloc, ErrorCode::E0658,
+	"top level or-patterns are not allowed for %<let%> bindings");
```

Comment:
We can't use E0658 here, as E0658 refers to the use of unstable features. In 1.49, top level or patterns were considered an unstable feature, which is why the compiler emitted E0658. It's now been stabilized. So the handling here needs to be a little different.

The error in itself does not have an error code. So we cannot have one either.

We then have to decide if we want to implement top level or-patterns (if they are needed by `core` or `std`, we'll need to implement them, but otherwise no) and go from there.

Furthermore, in Rust 1.49 with nightly features `let a | a = 15` is valid, but not in Rust 1.53 when toplevel or-patterns were stabilized. You need to use `let (a | a) = 15`.